### PR TITLE
CWL support

### LINKED ISSUE #27: config: add CWL config and worker
Once [REANA Workflow Engine CWL](https://github.com/reanahub/reana-workflow-engine-cwl) is ready we should add it to the current [config](https://github.com/reanahub/reana-resources-k8s/blob/master/reana_resources_k8s/templates/config.yaml#L26-L33) along with a new [worker](https://github.com/reanahub/reana-resources-k8s/blob/master/reana_resources_k8s/templates/config.yaml#L61-L78).

### PR DESCRIPTION
**Pull request by anton-khodak**
Closes #27

### CODE REVIEW

File: reana_resources_k8s/templates/deployments/workers-template.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ spec:
           value: tcp://zeromq-msg-proxy.default.svc.cluster.local:8666
         - name: QUEUE_ENV
           value: {{worker.WORKFLOW_ENGINE.NAME}}-{{worker.EXPERIMENT}}-queue
-        - name: REANA_WORKFLOW_ENGINE_YADAGE_EXPERIMENT
+        - name: REANA_WORKFLOW_ENGINE_EXPERIMENT
```
Comment (diegodelemos) [inline reana_resources_k8s/templates/deployments/workers-template.yaml:23]: we need to make a PR on `reana-workflow-engine-yadage` to update this env variable when this is merged

**Comment by diegodelemos** [comment]
For the record how to debug inside the cluster using `wdb` following [wdb docs](https://github.com/Kozea/wdb#docker):

1. You need some batteries on your containers so you can really debug. So add to your docker files this (preferably in the beginning of your Dockerfile so you avoid rebuilding this part that doesn't change):
```Dockerfile
...
RUN apt-get update && \
    apt-get install -y vim emacs-nox && \
    pip install wdb
ENV WDB_SOCKET_SERVER=wdb
ENV WDB_NO_BROWSER_AUTO_OPEN=True
ENV TERM=xterm
ENV FLASK_DEBUG=1
...
```

2. After this create your deployments as usual and add a `wdb` server deployment and service which can be created with `kubectl create -f  [filename]`:
```yaml
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: wdb
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: wdb
    spec:
      containers:
      - name: wdb
        image: kozea/wdb-server
        imagePullPolicy: IfNotPresent
        ports:
        - containerPort: 1984
        - containerPort: 19840
```
```yaml
---
apiVersion: v1
kind: Service
metadata:
  name: wdb
spec:
  type: "NodePort"
  ports:
   - port: 19840
     targetPort: 19840
     name: "socket"
     protocol: TCP
   - port: 1984
     targetPort: 1984
     name: "ui"
     protocol: TCP
  selector:
    app: wdb
```
3. Put a breakpoint:
```console
$ kubectl exec -ti [pod_name] bash
> emacs|vim /path/to/file/where/to/set/breakpoint.py
# Inside the file set the breakpoint as follows:
# import wdb; wdb.set_trace()
# save the file. Since we have FLASK_DEBUG active Flask will reload.
```

How to open debugger in browser:
```console
# get pod ip
$ minikube ip
192.168.99.100
$ kubectl describe service "wdb" | grep 'NodePort:'
NodePort:               socket  30337/TCP
NodePort:               ui      31839/TCP
# so we open http://192.168.99.100:31839
```

screen shot 2017-10-05 at 7 16 31 pm

screen shot 2017-10-05 at 7 18 57 pm

Limitations:
It is not possible for `celery` since the option `--autoreload` doesn't work and it is deprecated. To debug `celery` right now:
- Add `import wdb; wdb.set_trace()` to the beginning of the celery task 
- Build image
- Reinstantiate on kubernetes

This will bring up the breakpoint all the time but at least if you want to do code changes you can add them with `vim` or `emacs`

File: reana_resources_k8s/templates/deployments/workers-template.yaml (hunk 1 above)
Comment (anton-khodak) [reply reana_resources_k8s/templates/deployments/workers-template.yaml:23]: https://github.com/reanahub/reana-workflow-engine-yadage/pull/43

**Comment by anton-khodak** [comment]
@diegodelemos I suggest bringing your comment to http://reana.readthedocs.io/en/latest/ in "Contributing" section or similar

**Comment by diegodelemos** [comment]
closed via https://github.com/reanahub/reana-cluster/pull/47 since `reana-resources-k8s` is deprecated

**Pull request closed without merging**